code/game/objects/machinery/squad_supply/supply_beacon.dm

```diff
@@ -149,27 +136,76 @@
 		H.visible_message("[H] activates [src]",
 		"You activate [src]")
 
-/obj/item/squad_beacon/bomb/proc/deactivate(mob/living/carbon/human/H)
+/obj/item/squad_beacon/bomb/deactivate(mob/living/carbon/human/H)
 	var/delay = max(1 SECONDS, activation_time * 0.5 - 2 SECONDS * H.skills.getRating("leadership")) //Half as long as setting it up.
 	H.visible_message("<span class='notice'>[H] starts removing [src] from the ground.</span>",
 	"<span class='notice'>You start removing [src] from the ground, deactivating it.</span>")
-	if(do_after(H, delay, TRUE, src, BUSY_ICON_GENERIC))
-		message_admins("[ADMIN_TPMONTY(usr)] removed an orbital strike beacon.")
-		if(squad)
-			squad.squad_orbital_beacons -= src
-			squad = null
-		GLOB.active_orbital_beacons -= src
-		qdel(beacon_cam)
-		beacon_cam = null
-		activated = FALSE
-		anchored = FALSE
-		w_class = initial(w_class)
-		layer = initial(layer)
-		name = initial(name)
-		icon_state = initial(icon_state)
-		set_light(0)
-		playsound(src, 'sound/machines/twobeep.ogg', 15, 1)
-		H.visible_message("[H] deactivates [src]",
-		"You deactivate [src]")
-		H.put_in_active_hand(src)
-
+	if(!do_after(H, delay, TRUE, src, BUSY_ICON_GENERIC))
+		return
+	message_admins("[ADMIN_TPMONTY(usr)] removed an orbital strike beacon.")
+	if(squad)
+		squad.squad_orbital_beacons -= src
+		squad = null
+	GLOB.active_orbital_beacons -= src
+	qdel(beacon_cam)
+	beacon_cam = null
+	activated = FALSE
+	anchored = FALSE
+	w_class = initial(w_class)
+	layer = initial(layer)
+	name = initial(name)
+	icon_state = initial(icon_state)
+	set_light(0)
+	playsound(src, 'sound/machines/twobeep.ogg', 15, 1)
+	H.visible_message("[H] deactivates [src]",
+	"You deactivate [src]")
+	H.put_in_active_hand(src)
+
+/datum/component/supply_beacon
+	///Name of the supply beacon in the supply drop list
+	var/name = ""
+	///Where the supply drop is sent
+	var/turf/location
+	///If the supply beacon is shown on the supply drop list
+	var/registered = FALSE
+	/// Callback to tell the parent that the beacon was unregistered
+	var/datum/callback/unregistered_callback
+
+/datum/component/supply_beacon/Initialize(_unregistered_callback)
+	. = ..()
+	if(!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, COMSIG_REGISTER_BEACON, .proc/register)
+	unregistered_callback = _unregistered_callback
+	
+///Register this supply beacon into the supply beacon list
+/datum/component/supply_beacon/proc/register(datum/source, _name, turf/_location, life_time = 0 SECONDS)
+	SIGNAL_HANDLER
+	if(registered)
+		return
+	for(var/datum/component/supply_beacon/registered_beacon AS in GLOB.supply_beacon_list)
+		if(registered_beacon.name == _name)
+			return
```

Comment:
Why des this check exist

Reply:
To prevent people from having two antenna helmets in their bag and using them both. That leads to issue cause of similar names